src/System.Net.Security/tests/FunctionalTests/SslStreamAlertsTest.cs

```diff
@@ -0,0 +1,168 @@
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new SslStream(clientStream, true, AllowAnyServerCertificate))
+            using (var server = new SslStream(serverStream))
+            using (X509Certificate2 certificate = Configuration.Certificates.GetServerCertificate())
+            {
+                var handshake = new Task[2];
+
+                handshake[0] = server.AuthenticateAsServerAsync(certificate);
+                handshake[1] = client.AuthenticateAsClientAsync(certificate.GetNameInfo(X509NameType.SimpleName, false));
+
+                await Task.WhenAll(handshake).TimeoutAfter(TestConfiguration.PassingTestTimeoutMilliseconds);
+
+                var readBuffer = new byte[1024];
+
+                await server.ShutdownAsync();
+                int bytesRead = await client.ReadAsync(readBuffer, 0, readBuffer.Length);
+                // close_notify received by the client.
+                Assert.Equal(0, bytesRead);
+
+                await client.ShutdownAsync();
+                bytesRead = await server.ReadAsync(readBuffer, 0, readBuffer.Length);
+                // close_notify received by the server.
+                Assert.Equal(0, bytesRead);
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public async Task SslStream_StreamToStream_ClientInitiatedCloseNotify_Ok()
+        {
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new SslStream(clientStream, true, AllowAnyServerCertificate))
+            using (var server = new SslStream(serverStream))
+            using (X509Certificate2 certificate = Configuration.Certificates.GetServerCertificate())
+            {
+                var handshake = new Task[2];
+
+                handshake[0] = server.AuthenticateAsServerAsync(certificate);
+                handshake[1] = client.AuthenticateAsClientAsync(certificate.GetNameInfo(X509NameType.SimpleName, false));
+
+                await Task.WhenAll(handshake).TimeoutAfter(TestConfiguration.PassingTestTimeoutMilliseconds);
+
+                var readBuffer = new byte[1024];
+
+                await client.ShutdownAsync();
+                int bytesRead = await server.ReadAsync(readBuffer, 0, readBuffer.Length);
+                // close_notify received by the server.
+                Assert.Equal(0, bytesRead);
+
+                await server.ShutdownAsync();
+                bytesRead = await client.ReadAsync(readBuffer, 0, readBuffer.Length);
+                // close_notify received by the client.
+                Assert.Equal(0, bytesRead);
+            }
+        }
+        
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
```

Comment:
`PlatformID` has been renamed `TestPlatforms`. It would be good if it was updated here before merging. See #12284.